### PR TITLE
Changed the selected country setter to look for the iso2 property instead of object equality.

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
blackbaud/skyux-forms#192

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-selected-country-property-equality](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-selected-country-property-equality)

File: src/app/public/modules/country-field/country-field.component.ts [hunk 1]
```diff
@@ -372,6 +373,20 @@ export class SkyCountryFieldComponent implements ControlValueAccessor, OnDestroy
       });
   }
 
+  private checkNewCountryEquality(newCountry: SkyCountryFieldCountry) {
+    if (this._selectedCountry && newCountry) {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/country-field/country-field.component.ts:377]: You can write this more succinctly without all the nested `if`s:

```
if (country1 && country2) {
  return country1.iso2 === country2.iso2;
}

return country1 === country2;
```

File: src/app/public/modules/country-field/country-field.component.ts [hunk 2]
```diff
@@ -185,17 +185,18 @@ export class SkyCountryFieldComponent implements ControlValueAccessor, OnDestroy
         this.ngControl.control.markAsPristine();
       }
 
+      if (this.isInitialChange) {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/country-field/country-field.component.ts:188]: You can always set this to `false`. There's no need for the `if` statement here.

File: src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts [hunk 3]
```diff
@@ -95,4 +95,8 @@ export class CountryFieldReactiveTestComponent implements OnInit {
     return;
   }
 
+  public setValue(country: SkyCountryFieldCountry) {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts:98]: Remember to specify a return type.

File: src/app/public/modules/country-field/country-field.component.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: Good call. Removed

File: src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts]: Done

File: src/app/public/modules/country-field/country-field.component.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: Done. However, I did have to make a slight tweak to the last line which I outline the reasoning for in a code comment at that line.

File: src/app/public/modules/country-field/country-field.component.ts [hunk 4]
```diff
@@ -372,6 +373,20 @@ export class SkyCountryFieldComponent implements ControlValueAccessor, OnDestroy
       });
   }
 
+  private checkNewCountryEquality(newCountry: SkyCountryFieldCountry) {
```
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: Yup. That is a good point. That will allow us to reuse the method down the line if needed too. Changed.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-selected-country-property-equality](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-selected-country-property-equality)

**Pull request merged**